config/file.go

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) OpenFaaS authors 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package config
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"net/url"
+
+	"github.com/mitchellh/go-homedir"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	DefaultDir = "~/.openfaas"
+	// using .yml to avoid collision with openfaas-bitbar config
+	DefaultFile = "config.yml"
+)
+
+type ConfigFile struct {
+	AuthConfigs map[string]AuthConfig `json:"auths" yaml:"auths"`
+	FilePath    string                `json:"-" yaml:"-"`
+}
+
+type AuthConfig struct {
+	Auth  string `json:"auth,omitempty" yaml:"auth,omitempty"`
+	Token string `json:"token,omitempty" yaml:"token,omitempty"`
+}
+
+// New initializes a config file for the given file path
+func New(filePath string) (*ConfigFile, error) {
+	if filePath == "" {
+		return nil, errors.New("can't create config with empty filePath")
+	}
+	conf := &ConfigFile{
+		AuthConfigs: make(map[string]AuthConfig),
+		FilePath:    filePath,
+	}
+
+	return conf, nil
+}
+
+// EnsureFile creates the root dir and config file
+func EnsureFile() (string, error) {
```

Comment:
Would be cleaner when testing to allow the homedir to be overridden, currently you get your local `~/.openfaas/` dir polluted with test YAML.

Reply:
Yes - talked about this with @johnmccabe on Sunday. We should be using os.Tempdir or similar for integration tests.

As a side note about the cli tests, integration tests should perhaps be in their own area so we can run them independently.

Reply:
I've used `os.Tempdir` take a look at the latest commit.